src/status.ts

```diff
@@ -2,258 +2,385 @@ import * as vscode from 'vscode';
     const good = passing == total;
-    const icon = good ? 'check' : 'x';
     let testPassingText: string;
     if (total == 1) {
       testPassingText = localize('test.passing', '{0}/{1} test passing', passing, total);
     } else {
       testPassingText = localize('tests.passing', '{0}/{1} tests passing', passing, total);
     }
-    this._testButton.text = `$(${icon}) ${testPassingText}`;
-    this._testButton.color = good ? 'lightgreen' : 'yellow';
+    const icon = good ? 'check' : 'x';
+    return `$(${icon}) ${testPassingText}`;
   }
+}
+class BuildButton extends CheckCPPToolsButton {
+  private static readonly _build = localize('build', 'Build');
+  private static readonly _stop = localize('stop', 'Stop');
 
-  /** Reloads the content of the build button */
-  private _reloadBuildButton() {
-    this._buildButton.text = ``;
-    this._buildButton.text = this._isBusy ? `$(x) ${localize('stop', 'Stop')}` : `$(gear) ${localize('build', 'Build')}:`;
-    this._buildButton.command = this._isBusy ? 'cmake.stop' : 'cmake.build';
-    if (this._isBusy) {
-      this._buildButton.show();
-    }
-  }
+  command = 'cmake.build';
+
+  private _isBusy:boolean = false;
 
-  /**
-   * Whether or not to show a 'Build' or 'Stop' button. Changes the content
-   * of the button and the command that is executed when the button is pressed
-   */
-  private _isBusy: boolean = false;
-  setIsBusy(v: boolean) {
+  set isBusy(v: boolean) {
     this._isBusy = v;
-    this._reloadBuildButton();
+    this._button.command = v ? 'cmake.stop' : 'cmake.build';
+    this.update();
   }
 
-  private _reloadKitsButton() {
-    if (this._visible) {
-      if (this._activeKitName.length) {
-        this._kitSelectionButton.text = this._activeKitName;
-      } else {
-        this._kitSelectionButton.text = localize('no.kit.selected', 'No Kit Selected');
-      }
-      this.reloadVisibility();
-    } else {
-      this._kitSelectionButton.hide();
-    }
+  private _getCurrentText():string {
+    return this._isBusy?BuildButton._stop:BuildButton._build;
+  }
+  private _getCurrentIcon():string {
+    return this._isBusy?'$(x)':'$(gear)';
   }
 
-  setActiveKitName(v: string) {
-    if (v === SpecialKits.Unspecified) {
-      this._activeKitName = `[${localize('no.active.kit', 'No active kit')}]`;
-    } else {
-      this._activeKitName = v;
-    }
-    this._reloadKitsButton();
+  getTextIcon():string {
+    return this._getCurrentIcon();
   }
-  private _activeKitName: string = '';
+  getTextNormal():string {
+    return `${this._getCurrentIcon()} ${this._getCurrentText()}`;
+  }
+  getTooltipIcon():string {
+    return this._getCurrentText();
+  }
+  isVisible():boolean {
+    return this._isBusy || true;
+  }
+  protected getType():ButtonType {
+    return this._config.statusbar.build.type;
+  }
+}
+
+export class StatusBar implements vscode.Disposable {
+  private readonly _kitSelectionButton = new KitSelection(this._config, 3.6);
```

Comment:
It looks like the order of the status items was changed. Can you restore the original order?

Reply:
changed.